openstack/networking/v2/extensions/bgpvpns/results.go

```diff
@@ -0,0 +1,517 @@
+package bgpvpns
+
+import (
+	"net/url"
+
+	"github.com/gophercloud/gophercloud/v2"
+	"github.com/gophercloud/gophercloud/v2/pagination"
+)
+
+const (
+	invalidMarker = "-1"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// Extract is a function that accepts a result and extracts a BGP VPN resource.
+func (r commonResult) Extract() (*BGPVPN, error) {
+	var s BGPVPN
+	err := r.ExtractInto(&s)
+	return &s, err
+}
+
+func (r commonResult) ExtractInto(v any) error {
+	return r.Result.ExtractIntoStructPtr(v, "bgpvpn")
+}
+
+// BGPVPN represents an MPLS network with which Neutron routers and/or networks
+// may be associated
+type BGPVPN struct {
+	// The ID of the BGP VPN.
+	ID string `json:"id"`
+
+	// The user meaningful name of the BGP VPN.
+	Name string `json:"name"`
+
+	// Selection of the type of VPN and the technology behind it. Allowed
+	// values are l2 or l3.
+	Type string `json:"type"`
+
+	// Indicates whether this BGP VPN is shared across tenants.
+	Shared bool `json:"shared"`
```

Comment:
Again, [this field is undocumented](https://docs.openstack.org/api-ref/network/v2/#id834).

Reply:
It's a read-only field. It's set to true, when rbac rules are used.